File: day4/Day4.java
class Day4{
    public void moveZeroes(int[] nums) {
        
        int n = nums.length;
        int relativeChange = 0;
        
        for(int i = 0;i < n;i++) {
            if(nums[i] == 0) {
                relativeChange++;
            }
            else {
                nums[i-relativeChange] = nums[i];
            }
        }

        while(relativeChange > 0) {
            nums[n-relativeChange--] = 0;
        }
    }
}

File: day4/day4.cpp
class Solution {
public:
    void moveZeroes(vector<int>&A) {
    int N = A.size();
    int change = 0;
    
    for(int i = 0;i < N;i++) {
        if(A[i] == 0) {
            change++;
        }
        else {
            A[i - change] = A[i];
        }
    }
    while(change > 0) {
        A[N-change--] = 0;
    }
  }
};